geode-core/src/main/java/com/gemstone/gemfire/cache/RegionFactory.java

```diff
@@ -823,6 +825,38 @@ public void setPublisher(boolean v) {
     return getCache().createRegion(name, ra);
   }
   /**
+   * Creates a region with the given name in this factory's {@link Cache}
+   * using the provided configuration. Validation of the
+   * provided attributes may cause exceptions to be thrown if there are problems
+   * with the configuration data.
+   *
+   * @param name
+   *          the name of the region to create
+   * @param regionAttributes
+   *          the attributes for the region
+   * @param internalRegionArguments
+   *          additional arguments for the region
+   *
+   * @return the region object
+   * @throws LeaseExpiredException
+   *           if lease expired on distributed lock for Scope.GLOBAL
+   * @throws RegionExistsException
+   *           if a region, shared or unshared, is already in this cache
+   * @throws TimeoutException
+   *           if timed out getting distributed lock for Scope.GLOBAL
+   * @throws CacheClosedException
+   *           if the cache is closed
+   * @throws IllegalStateException
+   *           if the supplied RegionAttributes are incompatible with this region
+   *           in another cache in the distributed system (see
+   *           {@link AttributesFactory} for compatibility rules)
+   */
+  @SuppressWarnings("unchecked")
+  public Region<K,V> create(String name, RegionAttributes<K, V> regionAttributes, InternalRegionArguments internalRegionArguments)
```

Comment:
RegionFactory is an external API. So you can't add this "create" method to it because it takes "InternalRegionArguments" which is not part of the external API.